code/game/machinery/recharger.dm

```diff
@@ -41,17 +64,13 @@
 	if(!dropsafety(G))
 		return
 
-	var/allowed = 0
-	for (var/allowed_type in allowed_devices)
-		if (istype(G, allowed_type)) allowed = 1
-
-	if(allowed)
+	if(is_type_in_list(G, allowed_devices))
 		if(charging)
 			user << "<span class='warning'>\A [charging] is already charging here.</span>"
 			return
 		// Checks to make sure he's not in space doing it, and that the area got proper power.
 		if(!powered())
-			user << "<span class='warning'>The [name] blinks red as you try to insert the item!</span>"
+			user << "<span class='warning'>\The [name] blinks red as you try to insert the item!</span>"
 			return
 		if (istype(G, /obj/item/weapon/gun/energy/gun/nuclear) || istype(G, /obj/item/weapon/gun/energy/crossbow))
```

Comment:
You could throw that out aswell and overwrite get_cell for these guns with "return DEVICE_NO_CELL".

Reply:
Actually, it might not be intuitive, if you do that since you can insert the device into the charger, but it wont charge.
It would need a extra check when the something is inserted to see if get_cell returns a valid cell.
(And I shouldnt be commenting on pulls at midnight.)